ampersand-calendar.js

```diff
@@ -0,0 +1,290 @@
+      var weekday;
+      var week;
+      var data;
+      var selected;
+      var dateMoment;
+      for (var i = 0; i < startDayOfMonth + lastDateOfMonth; i++) {
+        var date = i - startDayOfMonth + 1;
+        dateMoment = moment(month).date(date);
+        weekday = i % 7;
+        week = Math.floor(i / 7);
+
+        var valid = false;
+        selected = false;
+
+        valid = i >= startDayOfMonth && (dateMoment.diff(now, 'days') < 0 || dateMoment.isSame(now, 'day'));
+
+        if (this.endDate !== null) {
+          selected = (this.startDate !== null && dateMoment.diff(this.startDate, 'days') >= 0) && dateMoment.diff(this.endDate, 'days') <= 0;
+        } else {
+          selected = (this.startDate !== null && dateMoment.diff(this.startDate, 'days') === 0);
+        }
+
+        data = {
+          valid: valid,
+          date: dateMoment,
+          selected: selected,
+          weekday: weekday,
+          week: week
+        };
+
+        dates.push(data);
+      }
+
+      weekday ++;
+      month.add(1, 'M');
+      for (var j = 0; j < (7 * (6 - week)) - weekday; j++) {
+        dateMoment = moment(month).date(j + 1);
+        selected = true;
+
+        if (this.endDate !== null) {
+          selected = (this.startDate !== null && dateMoment.diff(this.startDate, 'days') >= 0) && dateMoment.diff(this.endDate, 'days') <= 0;
+        } else {
+          selected = (this.startDate !== null && dateMoment.diff(this.startDate, 'days') === 0);
+        }
+
+        data = {
+          valid: false,
+          date: dateMoment,
+          selected: selected,
+          weekday: (j + weekday) % 7,
+          week: week + Math.floor((j + weekday) / 7)
+        };
+
+        dates.push(data);
+      }
+
+      return dates;
+    }
+  });
+
+  var CalendarView = AmpersandView.extend({
+    template: '<svg></svg>',
+    autoRender: true,
+    initialize: function() {
+      this.model._view = this;
+    },
+    events: {
+      'click .ampersand-calendar-month-previous': 'selectPreviousMonth',
+      'click .ampersand-calendar-month-next': 'selectNextMonth',
+      'click .ampersand-calendar-weekday-container': 'selectWeekday'
+    },
+    render: function() {
+      AmpersandView.prototype.render.call(this);
+
+      var calendar = this.svg = d3.select(this.el)
+        .attr('class', 'ampersand-calendar')
+        .attr('width', '100%')
+        .attr('height', '16em');
+
+      var month = calendar.append('g')
+        .attr('class', 'ampersand-calendar-month');
+
+      var monthBackground = month.append('rect')
+        .attr('class', 'ampersand-calendar-month-background')
+        .attr('width', '100%')
+        .attr('height', '2em')
+        .attr('rx', '1em')
+        .attr('ry', '1em');
+
+      var monthLabel = month.append('text')
+        .attr('class', 'ampersand-calendar-month-label')
+        .attr('x', '50%')
+        .attr('y', '1.75em')
+        .text(this.model.current.format('MMMM YYYY'));
```

Comment:
Right now you don't have that binding though, right?  It should always be possible to trigger that binding change however.  Even if `model.current.add()` doesn't trigger it directly (which it might), you can always trigger the `change:current` event yourself.